docs/developer-guide/ksqldb-clients/contributing.md

```diff
@@ -0,0 +1,112 @@
+|-----------|----------------------------------------|
+| STRING    | STRING                                 |
+| INTEGER   | INTEGER                                |
+| BIGINT    | BIGINT                                 |
+| DOUBLE    | DOUBLE                                 |
+| BOOLEAN   | BOOLEAN                                |
+| DECIMAL   | DECIMAL(4, 2)                          |
+| ARRAY     | ARRAY<INTEGER>                         |
+| MAP       | MAP<STRING, DOUBLE>                    |
+| STRUCT    | STRUCT<\`F1\` STRING, \`F2\` INTEGER>  |
+
+Array, map, and struct types may be recursively nested within each other, and may contain any of the other types as well.
+
+Users of the client may wish to receive result rows in different ways:
+- Do not block. Rather, perform an action asynchronously on each new row as it arrives
+- Block until the next row arrives
+- Block until all rows have arrived. This only makes sense in the context of pull queries and push queries with `LIMIT` clauses. 
+
+Details of client interfaces to support these methods will likely vary by language. For example, the [Java client](java-client.md)
+uses [Reactive Streams](http://www.reactive-streams.org/) for its asynchronous streaming interface. 
+
+### Terminating push queries ###
+
+Requests to terminate push queries should be sent to the [`/close-query` server endpoint](TODO).
+The client should expose an interface that allows users to pass in a push query ID to be passed to this endpoint.
+
+### Inserting new rows of data into existing ksqlDB streams ###
+
+ksqlDB supports inserting new rows of data into existing ksqlDB streams via the [`/inserts-stream` server endpoint](TODO).
+If the server successfully validates the target stream name from a request made to the `/inserts-stream` endpoint,
+the client may then write JSON-serialized rows of data to the connection, and the server will insert each
+asynchronously and write an ack containing the relevant sequence number to the response.
+
+Because the insertions happen asynchronously on the server side, it makes sense for the client to expose APIs
+for inserting one row at a time, but inserting batches of rows is less useful due to the lack of transactional guarantees.
+
+The client should provide a convenient way for users to create rows of data to insert.
+The Java client's [KsqlObject](TODO) is one example, though details will likely vary by language.
+
+### Creation of new streams, tables, and persistent queries ###
+
+These DDL/DML requests should be sent to the [`/ksql` server endpoint](../ksqldb-rest-api/ksql-endpoint.md).
+
+### Terminating persistent queries ###
+
+These requests should be sent to the [`/ksql` server endpoint](../ksqldb-rest-api/ksql-endpoint.md).
+
+### Listing existing streams, tables, topics, and queries ###
+
+These admin operation requests should be sent to the [`/ksql` server endpoint](../ksqldb-rest-api/ksql-endpoint.md).
+
+Configurations
```

Comment:
This part could use a little more substance. Maybe linking to an example in the Java client would help? I read this and Im not quite sure what to do as an implementer.

Reply:
Linked to the Java client API docs.